app/controllers/discourse_custom_topic_lists/lists_controller.rb

```diff
@@ -3,9 +3,49 @@
 module ::DiscourseCustomTopicLists
   class ListsController < ::ApplicationController
     requires_plugin PLUGIN_NAME
+    skip_before_action :check_xhr, only: [:lists_feed]
 
     def show
       render json: success_json
     end
+
+    def lists_feed
+      discourse_expires_in 1.minute
+
+      raise Discourse::NotFound if !SiteSetting.experimental_topics_filter
+      raise Discourse::InvalidParameters.new(:topic_list_name) if !params[:topic_list_name]
+      topic_query_opts = { no_definitions: !SiteSetting.show_category_definitions_in_topic_lists }
+
+      list_item =
+        DiscourseCustomTopicLists::CustomTopicList.new(list_target_user).find_by_slug(
+          params[:topic_list_name],
+        )
+      raise Discourse::NotFound if !list_item
+
+      topic_query_opts[:q] = list_item["query"]
+      %i[page q].each do |key|
+        if params.key?(key.to_s)
+          value = params[key]
+          raise Discourse::InvalidParameters.new(key) if !TopicQuery.validate?(key, value)
+          topic_query_opts[key] = value
+        end
+      end
+
+      @title = "#{SiteSetting.title} - #{list_item["name"]}"
+      @link = "#{Discourse.base_url}/lists/#{params[:topic_list_name]}"
+      @atom_link = "#{Discourse.base_url}/lists/#{params[:topic_list_name]}.rss"
+      @description = list_item["description"]
+      @topic_list = TopicQuery.new(list_target_user, topic_query_opts).list_filter
+
+      render "lists/list", formats: [:rss]
+    end
+
+    def list_target_user
+      if params[:user_id] && guardian.is_staff?
```

Comment:
I know we copied this from core but that code in core was written close to a decade ago. Is there a reason we need to allow a staff user to view the RSS feed of another user? I feel like that is something unnecessary at the expense of more complexity.

If we need it, we should add a test for this logic since it is borderline security related. We never want to accidentally allow a non-staff user to view the topic list of another user.

Reply:
I think it would be a "nice to have feature" if we end up having a user-based custom listing.

Currently, lists are group-based only, so I agree this is unnecessary.

I'll remove it!